tgui/packages/tgui/interfaces/BluespaceTap.js

```diff
@@ -26,20 +26,52 @@ export const BluespaceTap = (props, context) => {
     emagged,
     safeLevels,
     nextLevelPower,
+    autoShutown,
+    stabilizers,
+    overhead,
+    stabilizerPower,
   } = data;
   const barColor = (desiredLevel > inputLevel && 'bad') || 'good';
   return (
     <Window width={650} height={450}>
       <Window.Content scrollable>
         <Stack fill vertical>
-          {!!emagged && (
-            <NoticeBox danger={1}>Safety Protocols disabled</NoticeBox>
+          {!!emagged && <NoticeBox danger={1}>All safeties disabled</NoticeBox>}
+          {!autoShutown && !emagged && (
+            <NoticeBox danger={1}>Auto shutdown disabled</NoticeBox>
           )}
-          {!!(inputLevel > safeLevels) && (
-            <NoticeBox danger={1}>High Power, Instability likely</NoticeBox>
+          {!stabilizers && (
+            <NoticeBox danger={1}>Stabilizers disabled</NoticeBox>
+          )}
+          {inputLevel - overhead > safeLevels && !emagged ? (
+            <NoticeBox danger={1}>
+              Stabilizers overwhelmed, Instability likely
+            </NoticeBox>
+          ) : !emagged && inputLevel > safeLevels ? (
+            <NoticeBox>High Power, engaging stabilizers</NoticeBox>
+          ) : inputLevel > safeLevels ? (
+            <NoticeBox danger={1}>
+              Stabilizers disabled, Instability likely
+            </NoticeBox>
+          ) : (
+            ''
```

Comment:
this is a lot of spaghetti, could you abstract this out into it's own component? Instead of doing in-line conditions, you could just use some cleaner code and return the NoticeBox you want

Reply:
done